src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -80,6 +79,13 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
         {
             Stopwatch sw = Stopwatch.StartNew();
 
+            ServiceBundle.PlatformProxy.OtelInstrumentation?.LogActivity(new Dictionary<string, object> { 
```

Comment:
`OtelInstrumentation` is never null. It'll always get instantiated in AbstractPlatformProxy after it's called here for the first time.

In the scenario, when a platform doesn't support Otel, NullOtelInstrumentation is created with no-op implementation, however, these parameters and dictionaries are still allocated and methods are called unnecessarily. I would maybe just make OtelInstrumentation null in this case and do a null check? @bgavrilMS does this sound right, or is the compiler smart enough to not create these param objects if it sees that they are not used down the line?

Second scenario is similar but for platforms that support Otel but it's just not enabled. Is there a way to instantiate OtelInstrumentation only if it's enabled or have some boolean flag like we have for the ITelemetryClient?

Reply:
Updated